From 8af724a1c169abe92ea1605cf97c8c97e2cd8059 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Sun, 01 Mar 2020 08:10:15 +0200
Subject: [PATCH] [SW-8229] goya2: set debug interface to shared

When in pcie_recovery and debug interface is still enabled, we need to
interfaces to shared not mmu_bypass. This will make sure that only PSOC
areas defined by range register are accessable.

Change-Id: I026798ec0be7214ad8fa3607ba77db48e0ec58a0
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/security.c b/common/security.c
index 55e64d8..d61635b 100644
--- a/common/security.c
+++ b/common/security.c
@@ -106,12 +106,12 @@
 void security_debug_if_shared(void)
 {
 	/* Configure SMI,I2C and JTAG to be "shared" engine */
-	regmap_writel(MSTR_IF_AXUSER_HB_SEC_MMBP,
-		      mmI2C_S_MSTR_IF_AXUSER_HB_SEC);
-	regmap_writel(MSTR_IF_AXUSER_HB_SEC_MMBP,
-		      mmSMI_MSTR_IF_AXUSER_HB_SEC);
-	regmap_writel(MSTR_IF_AXUSER_HB_SEC_MMBP,
-		      mmJT_MSTR_IF_AXUSER_HB_SEC);
+	regmap_writel(I2C_S_MSTR_IF_AXUSER_CORE_SHARED_MASK,
+		      mmI2C_S_MSTR_IF_AXUSER_CORE);
+	regmap_writel(SMI_MSTR_IF_AXUSER_CORE_SHARED_MASK,
+		      mmSMI_MSTR_IF_AXUSER_CORE);
+	regmap_writel(JT_MSTR_IF_AXUSER_CORE_SHARED_MASK,
+		      mmJT_MSTR_IF_AXUSER_CORE);
 	regmap_writel(MSTR_IF_AXUSER_CORE_SHARED,
 		      mmPSOC_MSTR_IF_AXUSER_CORE);
 }
